paas_app_charmer/charm.py

```diff
@@ -143,6 +143,8 @@ def __init__(self, framework: ops.Framework, framework_name: str) -> None:
             self._on_secret_storage_relation_changed,
         )
         self.framework.observe(self.on.update_status, self._on_update_status)
+        if ops.JujuVersion.from_environ().has_secrets:
+            self.framework.observe(self.on.secret_changed, self._on_secret_changed)
```

Comment:
For what it's worth, you shouldn't need the guard here. If there are no Juju secrets, then you'll just get no secret events. ops makes all events available for observing regardless of what version of Juju the charm is executing under.

Reply:
I didn't know that. Thanks for the tip, updated it!